packages/payload/src/utilities/parseCookies.ts

```diff
@@ -1,13 +1,23 @@
 import type { Request } from 'express'
 
+import { APIError } from '../errors'
+
 export default function parseCookies(req: Request): { [key: string]: string } {
   const list = {}
   const rc = req.headers.cookie
 
   if (rc) {
     rc.split(';').forEach((cookie) => {
       const parts = cookie.split('=')
-      list[parts.shift().trim()] = decodeURI(parts.join('='))
+      const key = parts.shift().trim()
+      const encodedValue = parts.join('=')
+
+      try {
+        const decodedValue = decodeURI(encodedValue)
+        list[key] = decodedValue
+      } catch (e) {
+        throw new APIError(`Error decoding cookie value for key ${key}: ${e}`)
```

Comment:
Are the errors thrown from decodeURI going to show in a string? Sometimes errors have `error.message`. If we can't verify that toString for `${e}` will always work then we should just leave that off. The key is probably enough info to diagnose any problems.

Reply:
Looks like it throws an object in the shape of [URIError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/URIError#catching_an_urierror)

Reply:
If we really want to be safe here, we could just do:

```ts
try {
  const decodedValue = decodeURI(encodedValue)
  list[key] = decodedValue
} catch(e) {
  let message = typeof e === 'string' ? e : 'Unknown error'
  if (e instanceof URIError) {
    message = e.message
  }
  throw new APIError(`Error decoding cookie value for key ${key}: ${message}`)
}
```